mods/blocks/torch.lua

```diff
@@ -17,6 +17,22 @@ local function on_flood(pos, oldnode, newnode)
 	return false
 end
 
+local function on_flood_typed(pos, oldnode, newnode, torch_type)
```

Comment:
Duplicated function. Make `torch_type` be the full node name instead? That way you can apply to both the old torch and the new ones.